tests/Doctrine/Tests/ORM/Functional/OneToManyMatchingCriteriaTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Common\Collections\Criteria;
+use Doctrine\ORM\Mapping\FetchMode;
+use Doctrine\Tests\Models\ECommerce\ECommerceCategory;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * Tests sorting mechanism of Selectable::matching()
+ */
+class OneToManyMatchingCriteriaTest extends OrmFunctionalTestCase
+{
+    private $parent;
+    private $firstChild;
+    private $secondChild;
+    private $thirdChild;
+
+    protected function setUp() : void
+    {
+        $this->useModelSet('ecommerce');
+        parent::setUp();
+        $this->parent = new ECommerceCategory();
+        $this->parent->setName('Programming languages books');
+        $this->firstChild = new ECommerceCategory();
+        $this->firstChild->setName('Java books');
+        $this->secondChild = new ECommerceCategory();
+        $this->secondChild->setName('Php books');
+        $this->thirdChild = new ECommerceCategory();
+        $this->thirdChild->setName('BBQ books');
+    }
+
+    public function testMatchingWithNoOrderingUsesDatabaseDefault() : void
+    {
+        $this->createFixture();
+
+        $metadata = $this->em->getClassMetadata(ECommerceCategory::class);
```

Comment:
By providing a set of entities for the test we might avoid changing the metadata in runtime, which makes things slightly confusing 😁